presto-docs/src/main/sphinx/admin/tuning.rst

```diff
@@ -10,29 +10,392 @@ Config Properties
 
-* ``task.info-refresh-max-wait``:
-  Controls staleness of task information, which is used in scheduling. Increasing
-  this value can reduce coordinator CPU load, but may result in suboptimal
-  split scheduling.
-
-* ``task.max-worker-threads``:
-  Sets the number of threads used by workers to process splits. Increasing this number
-  can improve throughput, if worker CPU utilization is low, but will cause increased
-  heap space usage.
-
-* ``distributed-joins-enabled``:
-  Use hash distributed joins instead of broadcast joins. Distributed joins
-  require redistributing both tables using a hash of the join key. This can
-  be slower (sometimes substantially) than broadcast joins, but allows much
-  larger joins. Broadcast joins require that the tables on the right side of
-  the join fit in memory on each machine, whereas distributed joins only
-  need to fit in distributed memory across all machines. This can also be
-  specified on a per-query basis using the ``distributed_join`` session property.
-
-* ``node-scheduler.network-topology``:
-  Sets the network topology to use when scheduling splits. "legacy" will ignore
-  the topology when scheduling splits. "flat" will try to schedule splits on the same
-  host as the data is located by reserving 50% of the work queue for local splits.
++------------------------------------------+------------------------------------------+------------------------------------+-------------------------------------------------------+
++              property name               +                   type                   +           default value            +tunning the property                                   +
++==========================================+==========================================+====================================+=======================================================+
++      ``task.info-refresh-max-wait``      +          ``String`` (duration)           +             ``200 ms``             +Controls staleness of task information which is used   +
++                                          +                                          +                                    +in scheduling. Increasing this value can reduce        +
++                                          +                                          +                                    +coordinator CPU load but may result in suboptimal      +
++                                          +                                          +                                    +split scheduling.                                      +
++------------------------------------------+------------------------------------------+------------------------------------+-------------------------------------------------------+
++       ``task.max-worker-threads``        +               ``Integer``                +    ``Number of node CPU's * 4``    +Sets the number of threads used by workers to process  +
```

Comment:
We use java.lang.Runtime.availableProcessors for that and it does not document if it counts physical or logical cores. It really doesn't even explain if it means core or physical units. This is why I just paraphrased JavaDoc in order to avoid over interpretation of what we really do.
AFAIK this function returns logical cores in real configurations, but considering java architecture I would guess that it's platform specific what is provided as "processor" for JVM.

Neither Java Language Specification nor Java Virtual Machine Specification does explain what "available processors" are. My bet would be that it can change between operating system and JVM implementations as in normal circumstances user space application (which I consider JVM to be) are not aware of what hardware resources are assigned to them without asking OS about that. I also think that OS can change affinity of JVM process while it's already up and running and we don't update that value.

I considered short information more useful that this long explanation, but I can add it to a "tuning" column if you consider that useful.

Or am I wrong at some point?

Sources:
https://docs.oracle.com/javase/7/docs/api/java/lang/Runtime.html#availableProcessors()
https://docs.oracle.com/javase/specs/jls/se8/jls8.pdf
https://docs.oracle.com/javase/specs/jvms/se8/jvms8.pdf